app/livechat/server/lib/analytics/agents.js

```diff
@@ -9,8 +9,8 @@ const findAllAverageServiceTimeAsync = async ({
 		throw new Error('"start" and "end" must be provided');
 	}
 	return {
-		agents: await Users.findAllAverageServiceTime({ start, end, options }),
-		total: (await Users.findAllAverageServiceTime({ start, end })).length,
+		agents: await LivechatRooms.findAllAverageServiceTimeByAgents({ start, end, options }),
+		total: (await LivechatRooms.findAllAverageServiceTimeByAgents({ start, end })).length,
```

Comment:
My suggestion is to change the way we get the total number of records without pagination, I mean, we don't need to get all data from the DB just to know the `count/total` number, actually, we are wasting computational resources since we're allocation all documents in memory just to get the `length` value.
IMO, we can use a [native](https://docs.mongodb.com/manual/reference/operator/aggregation/count/) `MongoDB` feature that returns exactly the value we need. To achieve this we need to support an additional function parameter, allowing to set something like `onlyCount` parameter, also we'd need to return the `cursor` instead of `array`.